### PR TITLE
feat(WebTerminal): refactors code to useContext and custom hooks 

### PR DESCRIPTION
**Pull request by AlexanderMelox**
Contributes to #1415 

Refactored web terminal to use a custom hook and context to share state between the various web terminal components. This PR is in response to the release review question "Can we change `closeTerminal` to `onClose` or `onCloseTerminal` instead for consistency.".

This change altogether allows the user to now worry about passing an `onClose` or `open` prop to the web terminal and gives the user the liberty to do what they please when they call the hooks `closeWebTerminal, openWebTerminal, toggleWebTerminal`

### CODE REVIEW

File: packages/cloud-cognitive/src/components/WebTerminal/WebTerminal.js [hunk 1]
```diff
@@ -146,7 +146,7 @@ export let WebTerminal = React.forwardRef(
             hasIconOnly
             renderIcon={Close}
             kind="ghost"
-            iconDescription={closeIconDescription}
+            iconDescription="Close web terminal"
```
Comment (davidmenendez) [inline packages/cloud-cognitive/src/components/WebTerminal/WebTerminal.js:149]: why did you remove the `closeIconDescription` prop?

**Review by davidmenendez — COMMENTED** [review]
looks good to me. i just had one comment about the close icon label.

File: packages/cloud-cognitive/src/components/WebTerminal/WebTerminal.js (hunk 1 above)
Comment (AlexanderMelox) [reply packages/cloud-cognitive/src/components/WebTerminal/WebTerminal.js:149]: Thinking about it, what else can this be called? The literal action is closing the web terminal, I don't think designers would really want to change this to something else? Thoughts?

**Comment by AlexanderMelox** [comment]
Some new changes with the last two commits. 

1. Update props
Added a new prop `isInitiallyOpen` this allows the user to make the WebTerminal open on load. This prop also updates the custom hook to reflect the state change.

2. Updated the tests and added new ones to test the functionality of the `useWebTerminal()` hook.
3. Added `@testing-library/react-hooks` to be able to test react custom hooks in jest.
We need this library to use hooks since we cannot use hooks directly in the test function since it's not a react component.

**Comment by AlexanderMelox** [comment]
New updates: 

1. Added the WebTerminalProvider to the canary flag enabled test. 
2. Added WebTerminalProvider to package settings

**Comment by matthewgallo** [comment]
Just posting an update here from our discussion around this from earlier today, I think this approach will be ready to merge once changes are made in `index-all-enabled.test.js` to scope the `WebTerminalProvider` to only wrap around the `WebTerminal` component.

**Review by matthewgallo — APPROVED** [review]
